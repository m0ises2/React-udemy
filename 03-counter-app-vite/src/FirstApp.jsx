import PropTypes from "prop-types";

export const FirstApp = ( { title, subtitle, age } ) => {
    return (
        <>
            <h1 data-testid='test title'> { title } </h1>
            <p data-testid='test subtitle'> { subtitle } </p>
            <p> { age } years old </p>
        </>
    ) // Esto es un fragmento
}

FirstApp.propTypes = {
    age: PropTypes.number.isRequired,
    subtitle: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
};

FirstApp.defaultProps = {
    age: 35,
    subtitle: 'Subtitulo por defecto'
};
