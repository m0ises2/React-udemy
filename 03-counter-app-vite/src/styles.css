html, body {
    background-color: rgb(0, 0, 0);
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 15px;
    font-size: 1.3rem;
}