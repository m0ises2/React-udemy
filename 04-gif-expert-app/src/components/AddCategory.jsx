import { useState } from "react"
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setinputValue] = useState('');

    const onInputChange = ({ target: { value: inputValue } } ) => {
        setinputValue(inputValue)
    };

    const onFormSubmit = (event) => {
        event.preventDefault();

        if (inputValue.trim().length <= 1) return;

        onNewCategory(inputValue.trim());

        setinputValue('');
    };

    return (
        <form 
            aria-label="form"
            onSubmit={ onFormSubmit }>
            <input
                type="text"
                placeholder = "Buscar gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />            
        </form>
    );
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
