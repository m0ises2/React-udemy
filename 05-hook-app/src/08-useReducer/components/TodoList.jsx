import PropTypes from "prop-types";
import { TodoItem } from './TodoItem';

export const TodoList = ({ todos, onDeleteTodo, onToggleTodo }) => {
  return (
    <ul className='list-group'>
      {
        todos.map(todo => (
          <TodoItem
            key={ todo.id }
            id={ todo.id }
            description={ todo.description }
            done={ todo.done }
            onDeleteTodo={onDeleteTodo}
            onToggleTodo={onToggleTodo}
          />
        ))
      }
    </ul>
  )
};

TodoList.propTypes = {
    todos: PropTypes.array.isRequired
};
