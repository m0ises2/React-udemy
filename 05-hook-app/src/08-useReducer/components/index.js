export * from './AddTodo';
export * from './TodoItem';
export * from './TodoList';