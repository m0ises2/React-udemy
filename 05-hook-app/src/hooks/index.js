export * from './UseCounter';
export * from './useFetch';
export * from './useForm';