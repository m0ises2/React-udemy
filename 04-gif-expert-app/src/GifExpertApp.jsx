import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);
  
  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory.toLowerCase())) return;
    
    setCategories([newCategory.toLowerCase(),...categories]);
  };

  return (
    <>
      {/* Titulo */}
      <h1>GitExpertApp</h1>

      {/* Input */}
      <AddCategory 
        onNewCategory={ onAddCategory }
      />
      
      {/* Lista de Gif */}
      { 
        categories.map(category => (
          <GifGrid 
            key={ category }
            category={ category }
          />
        ))
      }
        {/* Gif Item */}

    </>
  );
};
