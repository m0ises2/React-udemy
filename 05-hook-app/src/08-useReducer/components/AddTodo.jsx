import PropTypes from "prop-types";
import { useForm } from "../../hooks";

export const AddTodo = ({ onNewTodo }) => {
  const { description, onInputChange, onReset } = useForm({
    description: ''
  });

  const onSubmit = (event) => {
    event.preventDefault();

    if (description.trim().length <= 1) return;

    onNewTodo({
      id: new Date().getTime(),
      description,
      done: false
    });

    onReset();
  };

  return (
    <form
        onSubmit={ onSubmit }
    >
        <input 
          type="text" 
          placeholder='QuÃ© hay que hacer?' 
          className='form-control'
          name="description"
          value={ description }
          onChange={ onInputChange }
        />  
    <   button type='submit' className='btn btn-outline-primary mt-1'>  Agregar </button>
    </form>
  )
}

AddTodo.propTypes = {
    onNewTodo: PropTypes.func.isRequired
}