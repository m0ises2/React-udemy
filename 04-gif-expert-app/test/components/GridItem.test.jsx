import { render, screen } from '@testing-library/react';
import { GridItem } from "../../src/components";

describe('pruebas de componente GridItem', () => {
    const url = 'https://www.google.com/';
    const title = 'This is a mock title';

    test('debe hacer match con el snapshot', () => {
        const { container } = render(
            <GridItem
                title = { title }
                url = { url }
            />
        );

        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el URL y el ALT indicados', () => {
        render(
            <GridItem
                title = { title }
                url = { url }
            />
        );

        const { alt, src } = screen.getByRole('img');

        expect(alt).toBe(title);
        expect(src).toBe(url);
    });

    test('debe mostrar el titutlo en el componente', () => {
        render(
            <GridItem
                title = { title }
                url = { url }
            />
        );

        expect(screen.getByText(title)).toBeTruthy();
    });
});

