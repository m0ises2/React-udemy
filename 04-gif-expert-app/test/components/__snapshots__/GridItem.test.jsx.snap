// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas de componente GridItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="This is a mock title"
      src="https://www.google.com/"
    />
    <p>
       
      This is a mock title
       
    </p>
  </div>
</div>
`;
